File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    count = 0      
    zeros_removed = 0
    
    while s != "1":
        count += 1
        num_zeros = s.count('0')
        zeros_removed += num_zeros
        ones_count = len(s) - num_zeros
        s = bin(ones_count)[2:]
        
    return [count, zeros_removed]

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>

using namespace std;

string toBinary(int n) {
    string res = "";
    while (n > 0) {
        res = to_string(n % 2) + res;
        n /= 2;
    }
    return res;
}

vector<int> check(int count, int remove, string s) {
    if (s == "1") {
        return {count, remove};
    }

    string str = ""; 

    for (int i = 0; i < s.size(); i++) {
        char ch = s[i]; 
        if(ch == '0') {   
            remove++;
        } else {
            str += ch; 
        }
    }
    
    count++;
    
    string next_s = toBinary(str.size());
    
    return check(count, remove, next_s); 
}

vector<int> solution(string s) {
    return check(0, 0, s);
}